lib/galaxy/model/migrate/versions/util.py

```diff
@@ -86,17 +86,22 @@ def add_column(column, table, metadata, **kwds):
     :type metadata: :class:`Metadata`
     """
     try:
+        index_to_create = None
         migrate_engine = metadata.bind
         if not isinstance(table, Table):
             table = Table(table, metadata, autoload=True)
         if migrate_engine.name == 'sqlite' and column.index and column.foreign_keys:
             # SQLAlchemy Migrate has a bug when adding a column with both a
             # ForeignKey and an index in SQLite. Since SQLite creates an index
             # anyway, we can drop the explicit index creation.
+            # TODO: this is hacky, but it solves this^ problem. Needs better solution.
+            index_to_create = (kwds['index_name'], table, column.name)
             del kwds['index_name']
             column.index = False
         column.create(table, **kwds)
         assert column is table.c[column.name]
+        if index_to_create:
```

Comment:
Did you check if just removing the the if on line 93 is enough ?

Reply:
Yep, was the first thing I did - many things broke ("no such table" errors, etc.) I didn't look in detail though - just fixed it.